src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -149,8 +149,9 @@ private bool IsSupportedKeyWrapAlgorithm(string algorithm, SecurityKey key)
         /// </summary>
         /// <param name="algorithm">the algorithm to check.</param>
         /// <param name="key">the <see cref="SecurityKey"/>.</param>
+        /// <param name="isPrivateKey">Whether the <see cref="SecurityKey"/> is private key then set this to true.</param>
         /// <returns>true if 'algorithm, key' pair is supported.</returns>
-        public virtual bool IsSupportedAlgorithm(string algorithm, SecurityKey key)
+        public virtual bool IsSupportedAlgorithm(string algorithm, SecurityKey key, bool isPrivateKey)
```

Comment:
You're changing the parameter list for a public virtual method on a class that's public and pre-existing. This is definitely a breaking change. You can't change the signature on a method that we've already shipped that's public virtual without breaking people who used the old signature. Please either figure out how to do this without isPrivateKey or you'll need to add a new method with that parameter.